crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -0,0 +1,281 @@
+const RAW_CHAIN_SPEC_FILE_NAME: &str = "raw-parachain-chainspec.json";
+
+#[derive(Clone, ValueEnum, Default, VariantArray, Eq, PartialEq)]
+/// Supported chain types for the resulting chain spec.
+pub(crate) enum ChainType {
+	// A development chain that runs mainly on one node.
+	#[default]
+	Development,
+	// A local chain that runs locally on multiple nodes for testing purposes.
+	Local,
+	// A live chain.
+	Live,
+}
+
+impl ChainType {
+	fn into_str(self) -> &'static str {
+		match self {
+			ChainType::Development => "Development",
+			ChainType::Local => "Local",
+			ChainType::Live => "Live",
+		}
+	}
+}
+
+#[derive(Clone, ValueEnum, Default, VariantArray, Eq, PartialEq)]
+/// Supported relay chains that can be included in the resulting chain spec.
+pub(crate) enum RelayChain {
+	Kusama,
+	KusamaLocal,
+	Rococo,
+	RococoLocal,
+	Paseo,
+	#[default]
+	PaseoLocal,
+	Polkadot,
+	PolkadotLocal,
+}
+
+impl RelayChain {
+	fn into_str(self) -> &'static str {
+		match self {
+			RelayChain::Kusama => "kusama",
+			RelayChain::KusamaLocal => "kusama-local",
+			RelayChain::Rococo => "rococo",
+			RelayChain::RococoLocal => "rococo-local",
+			RelayChain::Paseo => "paseo",
+			RelayChain::PaseoLocal => "paseo-local",
+			RelayChain::Polkadot => "polkadot",
+			RelayChain::PolkadotLocal => "polkadot-local",
+		}
+	}
+}
+
+#[derive(Args)]
+pub struct BuildSpecCommand {
+	/// Directory path for your project [default: current directory].
+	#[arg(long)]
+	pub(crate) path: Option<PathBuf>,
+	/// For production, always build in release mode to exclude debug features.
+	#[clap(short, long, default_value = "true")]
+	pub(crate) release: bool,
+	/// Parachain ID to be used when generating the chain spec files.
+	#[arg(short = 'i', long = "id")]
+	pub(crate) id: Option<u32>,
+	/// Whether to keep localhost as a bootnode.
+	#[clap(long, default_value = "true")]
+	pub(crate) default_bootnode: bool,
+	/// Type of the chain [default: Development].
+	#[arg(short = 't', long = "type", value_enum)]
+	pub(crate) chain_type: Option<ChainType>,
+	/// Relay chain this parachain will connect to [default: PaseoLocal].
+	#[arg(long, value_enum)]
+	pub(crate) relay: Option<RelayChain>,
+	// Deprecation flag, used to specify whether the deprecation warning is shown.
+	#[clap(skip)]
+	pub(crate) valid: bool,
+}
+
+impl BuildSpecCommand {
+	/// Executes the command.
+	pub(crate) async fn execute(self) -> anyhow::Result<&'static str> {
+		// If para id has been provided we can build the spec
+		// otherwise, we need to guide the user.
+		let _ = match self.id {
+			Some(_) => self.build(&mut cli::Cli),
+			None => {
+				let config = guide_user_to_generate_spec().await?;
+				config.build(&mut cli::Cli)
+			},
+		};
+		Ok("spec")
+	}
+
+	/// Builds a parachain spec.
+	///
+	/// # Arguments
+	/// * `cli` - The CLI implementation to be used.
+	fn build(self, cli: &mut impl cli::traits::Cli) -> anyhow::Result<&'static str> {
```

Comment:
Does it make sense to check if it's a parachain before attempting the build with the `spec` command? I tried with a contract, and it only showed an error after the entire process was completed.

We have already the method: https://github.com/r0gue-io/pop-cli/blob/main/crates/pop-parachains/src/build.rs#L43 and the example on when is called: https://github.com/r0gue-io/pop-cli/blob/main/crates/pop-cli/src/commands/build/mod.rs#L65

Reply:
As for the last commit, it will silently exit without triggering any extra execution if the project is not a supported type.

Reply:
I see!  You added the code there already: https://github.com/r0gue-io/pop-cli/pull/257/files#diff-5ab1db70ee18975f80c440aa7fd49171bee41eda0fef6120ac04830d8d1b908fR154 will only be to add an error if is not supported: `cli::Cli.outro_cancel("🚫 Error.")?;`

Reply:
Appreciated :+1: